types/webpack/index.d.ts

```diff
@@ -54,7 +60,7 @@ declare namespace webpack {
          *   <li>"atom" Compile for usage in electron (formerly known as atom-shell), supports require for modules necessary to run Electron.</li>
          * <ul>
          */
-        target?: string;
+        target?: 'web' | 'webworker' | 'node' | 'async-node' | 'node-webkit' | 'atom';
```

Comment:
It would be great if you could make this `target` type much more precise for all its variations...

Documents can be found [here](https://webpack.js.org/configuration/target/)

Also, change comments above if needed. Thank you~